pkg/monitor/agent/listener1_2.go

```diff
@@ -86,6 +86,5 @@ func (ml *listenerv1_2) Version() listener.Version {
 func (ml *listenerv1_2) Close() {
 	ml.once.Do(func() {
 		ml.conn.Close()
-		close(ml.queue)
```

Comment:
If we are removing this close, when will the go routine initialized on line 44 terminate?

Reply:
fair enough, i'll try harder.